cstrml/market/src/lib.rs

```diff
@@ -63,18 +63,24 @@ pub struct Provision<Hash> {
     pub file_map: BTreeMap<MerkleRoot, Hash>,
 }
 
+pub type BalanceOf<T> =
+    <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+
 /// An event handler for paying market order
-pub trait Payment<AccountId, Hash> {
+pub trait Payment<AccountId, Hash, Balance> {
     // Pay the storage order, return an UNIQUE `transaction id`🙏🏻
     fn pay_sorder(client: &AccountId, provider: &AccountId, value: Balance) -> Hash;
+
+    // Start delayed pay
+    fn start_delayed_pay(sorder_id: &Hash);
 }
 
 impl<T: Trait> Payment<<T as system::Trait>::AccountId,
-    <T as system::Trait>::Hash> for Module<T>
+    <T as system::Trait>::Hash, BalanceOf<T>> for Module<T>
 {
     fn pay_sorder(client: &<T as system::Trait>::AccountId,
```

Comment:
If we implement in `payment` module, we can remove this implementation right?

Reply:
yes. We can remove this one. But market's tests don't include payment module right now. I prefer having this one.

Reply:
Then move this impl into test?